src/frame.js

```diff
@@ -0,0 +1,50 @@
+function frame() {
+  this._firstScore = null;
```

Comment:
It feels like an array might be a more suitable structure here than separate instance variables.